Assets/Scripts/UI/DialogBox/LUA/DialogBoxLua.cs

```diff
@@ -0,0 +1,159 @@
+﻿#region License
+// ====================================================
+// Project Porcupine Copyright(C) 2016 Team Porcupine
+// This program comes with ABSOLUTELY NO WARRANTY; This is free software, 
+// and you are welcome to redistribute it under certain conditions; See 
+// file LICENSE, which is part of this source code package, for details.
+// ====================================================
+#endregion
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Xml;
+using System.Xml.Serialization;
+using MoonSharp.Interpreter;
+using UnityEngine;
+using UnityEngine.UI;
+
+[MoonSharpUserData]
+public class DialogBoxLua : DialogBox
+{
+    private string title;
+
+    private object extraData;
+
+    private EventActions events;
+
+    public Transform Content { get; protected set; }
+
+    public DialogBoxResult Result { get; set; }
+
+    public string Title
+    {
+        get
+        {
+            return title;
+        }
+
+        protected set
+        {
+            title = value;
+            transform.GetChild(0).GetChild(0).GetComponentInChildren<Text>().text = value;
+        }
+    }
+    
+    public override void ShowDialog()
+    {
+        base.ShowDialog();
```

Comment:
Suggestion: I would like a way to reference show dialog in addition to  close dialog from lua. that seems like a very likely thing to need, say if you wanted to have a sound play when a dialog opens.